app/src/main/java/ch/epfl/balelecbud/util/intents/FlowUtil.java

```diff
@@ -121,26 +118,13 @@ public static Intent packCallback(Slot[] slots, @NonNull Intent intent) {
      * @param intent the intent to retrieve the array from
      * @return       a list of the slots stored in the intent or null if the Intent wasn't properly formatted
      */
-    public static List<Slot> unpackCallback(@NonNull Intent intent) {
+    public static ArrayList<Slot> unpackCallback(@NonNull Intent intent) {
```

Comment:
Why use ArrayList instead of List in the signature ?

Reply:
Because `Bundle` (and so `Intent`) requires `ArrayList`

Reply:
Ok that's weird, but not our problem. Closing that conversation